### PR TITLE
Minimal fish shell completions

### PR DESCRIPTION
**Pull request by ev-dev**
### Initial Shell Completions
For fish shell only; contains completions for flags but does not include flag sub-options like language lists or subcommands.

From what I understand #482 took place because of bugs with completion autogeneration. I added the 'completions' directory to assets so that perhaps others will follow with ported or manually written bash/zsh completions.

### CODE REVIEW

**Comment by cben** [comment]
Tested.  Basically works.
Does not suffer from #372 problem — I can complete file names everywhere.   Including places where strictly I shouldn't, e.g. <code>bat --theme <kbd>TAB</kbd></code> which ideally would complete theme names not files.  But perfect is the enemy of good...

**Comment by sharkdp** [comment]
Thank you very much for your contribution!

My only "concern" with this is that it has to be maintained manually. Most contributors will probably not update the completions files when they change command-line options or change help texts (the same happens with the man page). This means that I have to remind people in the PRs or do it myself when I create a new release.

I'm inclined to merge this, but I'm not sure if I will always find the time to keep it up-to-date.

**Comment by sharkdp** [comment]
Oh, before I merge this, we should definitely make sure that the completion file will be added to the release tarballs and Debian packages. See #391 for details.

File: bat.fish [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Fish Shell Completions (WIP)
+
+# Place or symlink to $XDG_CONFIG_HOME/fish/completions/bat.fish ($XDG_CONFIG_HOME is usually set to ~/.config)
```
Comment (sharkdp) [inline bat.fish]: I think this can be removed if we properly install this file?

File: bat.fish [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+complete -c bat -s l -l language -d "Set the language for syntax highlighting"
+
+complete -c bat -l list-languages -d "Display list of supported languages for syntax highlighting"
+
+complete -c bat -s m -l map-syntax -d "<from:to> Map a file extension or file name to an existing syntax"
+
+# TODO: add completion for theme list
```
Comment (sharkdp) [inline bat.fish]: `bat --list-themes` does produce a newline-separated list of theme names if it is run in non-interactive mode, i.e. if the output is piped somewhere.

**Comment by ev-dev** [comment]
I definitely understand the concern regarding future contributor updates @sharkdp, manually written completions would hopefully be a temporary workaround until some future upgrade of the autogeneration piece. 

Until then, It might be worth adding a note in the readme or a quick CONTRIBUTING.md with some guidelines, I'd add it to this PR but I don't want to impose in that regard.

**Comment by ev-dev** [comment]
I pushed up some minor updates including working --theme autocompletion after that helpful note on piped --list-themes output

I tried getting the completions as part of release packages during testing but I don't have a lot of experience with cargo builds or how to include completions as part of an install for linux/windows (could probably figure out macOS tho if it just involves updating a brew formula or something)

**Comment by sharkdp** [comment]
Thank you very much for the updates!

The deployment should hopefully work just like this.

**Comment by sharkdp** [comment]
Can someone from this thread maybe help with actually shipping the fish completions in the Debian package (#554)?

**Pull request merged**